### PR TITLE
Add CI tests for java_cert and java_keystore

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
More tests in CI \o/

##### ISSUE TYPE
- Test Pull Request

##### COMPONENT NAME
java_cert
java_keystore

### CODE REVIEW

**Comment by felixfontein** [comment]
There are quite a few things done by the java_keystore module that can be improved/fixed:
1. it writes everything to fixed filenames in /tmp/, and actually keeps the PKCS12 temp file it creates there.
2. it should support providing path names to certificate and private key, since it needs them written to disk anyway.
3. passwords to `keytool` are provided as command line arguments. This is already implemented correctly for the `openssl` call (done in when it was implemented #276; it uses stdin).

**Comment by felixfontein** [comment]
I created bug reports/feature requests (#1667, #1668, #1669, #1670) for improvements of java_keystore.

**Review by tadeboro — APPROVED** [review]
Thank you, @felixfontein, for your contribution.

As far a the functionality and logic go, I have nothing to complain about. I did add a few suggestions on the structure of integration tests but feel free to ignore them if you disagree with my personal preferences.

File: tests/integration/targets/java_keystore/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        certificate: "{{lookup('file', output_dir ~ '/' ~ item.name ~ '.pem') }}"
+        private_key: "{{lookup('file', output_dir ~ '/' ~ (item.keyname | default(item.name)) ~ '.key') }}"
+        private_key_passphrase: "{{ item.passphrase | default(omit) }}"
+        password: changeit
+        dest: "{{ output_dir ~ '/' ~ item.name ~ '.jks' }}"
+      loop:
+        - name: cert
+        - name: cert-pw
+          passphrase: hunter2
+      register: result_idem
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply tests/integration/targets/java_keystore/tasks/main.yml]: I usually try to avoide these features of YAML, and use `set_fact` instead, but I agree that for such tests, YAML anchors are a good thing.

[the remark this answers is not in the document]
Comment (felixfontein) [reply tests/integration/targets/java_keystore/tasks/main.yml]: (I definitely should use them more often...)

File: tests/integration/targets/java_keystore/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      assert:
+        that:
+          - result is changed
+          - result_check is changed
+          - result_idem is not changed
+          - result_idem_check is not changed
+          - result_change is changed
+          - result_change_check is changed
+          # - result_pw_change is changed        # FIXME: module currently crashes
+          # - result_pw_change_check is changed  # FIXME: module currently crashes
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply tests/integration/targets/java_keystore/tasks/main.yml:172]: For me it depends on the tests. I usually keep one assert together for some related tests, where related tests usually are something like create / idempotence / change (like here). If there are multiple such sequences, every one gets its own assert. In this case, there's only one.

File: tests/integration/targets/java_keystore/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    - name: Create a Java key store for the given certificates (password changed)
+      community.general.java_keystore:
+        <<: *create_key_store_data
+        password: hunter2
+      loop: *create_key_store_loop_new_certs
+      register: result_pw_change
+      when: false  # FIXME: module currently crashes
+
+    - name: Validate reults
```
Comment (felixfontein) [inline tests/integration/targets/java_keystore/tasks/main.yml:127]:
```suggestion
    - name: Validate results
```

**Comment by felixfontein** [comment]
@tadeboro thanks for reviewing this!

**Pull request merged**